### PR TITLE
vm create: expose --boot-diagnostics-storage to capture boot log

### LINKED ISSUE #5634: Adding Boot diagnostics option in VM create
---

### Environment summary

Install Method (e.g. pip, interactive script, apt-get, Docker, MSI, edge build) / CLI version (`az --version`) / OS version / Shell Type (e.g. bash, cmd.exe, Bash on Windows)  

```
```

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #5634 
Related command flows will be
```bash
    1. az storage account create -g yg -n ygstor --sku standard_lrs
    2. az vm create -g yg -n yg-1  --image ubuntults --boot-diagnostics-storage ygstor
```
For VM with unmanaged disk, I thought about defaulting the storage account to the os disk's but it turns out not very worthy for 2 reasons:
1. boot diagnostics only works with standard storage sku, so CLI default on using premium sku will not work.
2. VMs using managed disk is the majority scenario now which requires a pre-existing storage account any way. So defaulting for vm with `unmanaged disk` with `standard_lrs` will be pretty minority, hence could become confusing

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6302
This is an **experimental preview** for @microsoft users.

**Comment by colemickens** [comment]
Can you make this private build public? I need this functionality for early stage boot diagnostics. I'm not getting anywhere with trying to Stop, enable diagnostics, and reboot. For whatever reason, I can't even get Azure to Stop the VM I'm trying to debug.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -155,6 +155,8 @@ def load_arguments(self, _):
         c.argument('application_security_groups', resource_type=ResourceType.MGMT_NETWORK, min_api='2017-09-01', nargs='+', options_list=['--asgs'], help='Space-separated list of existing application security groups to associate with the VM.', arg_group='Network', validator=validate_asg_names_or_ids)
         c.argument('write_accelerator', nargs='*', min_api='2017-12-01', arg_group='Storage',
                    help="enable/disable disk write accelerator. Use singular value 'true/false' to apply across, or specify individual disks, e.g.'os=true 1=true 2=true' for os disk and data disks with lun of 1 & 2")
+        c.argument('boot_diagnostics_storage',
+                   help='pre-existing storage account name or its blob uri to capture boot diagnostics. Its sku should be one of Standard_GRS, Standard_LRS and Standard_RAGRS')
```
Comment (troydai) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Does the service need permission to the storage account?

Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: No extra permission configuration is needed. I believe CRP just accesses the storage under VM creator's identity captured through the original token from the VM PUT

**Comment by yugangw-msft** [comment]
@dsboulder, the release date will be around May 22, based on the milestone page https://github.com/Azure/azure-cli/milestones

**Comment by yugangw-msft** [comment]
storage name has to be unique across groups and subscriptions, so accepting a name is  enough

**Pull request merged**